src/zenml/zen_server/routers/stacks_endpoints.py

```diff
@@ -64,10 +64,14 @@ def list_stacks(
     stacks: List[StackResponseModel] = []
 
     # Get private stacks unless `is_shared` is set to True
-    if is_shared is None or not is_shared:
+    if (
+        is_shared is None
+        or not is_shared
+        and user_name_or_id == auth_context.user.id
```

Comment:
This won't work for `list_stacks(user_name_or_id=None)` which is the default `list_stacks()` call right?
```suggestion
        and (not user_name_or_id or user_name_or_id == auth_context.user.id)
```

Reply:
good point, changed it everywhere